External/FEXCore/Source/Interface/IR/AOTIR.cpp

```diff
@@ -200,6 +198,9 @@ namespace FEXCore::IR {
     }
 
     for (;;) {
+      // This code is tricky to refactor so it doesn't allocate memory through glibc.
+      FEXCore::Allocator::YesIKnowImNotSupposedToUseTheGlibcAllocator glibc;
```

Comment:
The entire std::function object allocates memory and deallocates once the object reaches end of scope.

Adding comment.

Reply:
Are you sure the function itself allocates memory here? It's using move construction, so the allocation should already have happened when appending to AOTIRCaptureCacheWriteoutQueue.

For deallocation you're right though. And it means we can't move the scope-guard down any further either since it needs to live longer than `fn` :(

So the comment needs a minor update for this unless I'm missing something else about the allocation: `The moved std::function object deallocates memory at the end of scope.` would be fine.